crates/replication/src/lib.rs

```diff
@@ -272,21 +271,18 @@ impl Replicator {
         }
 
         let frames = client
-            .log_entries(pb::LogOffset {
+            .batch_log_entries(pb::LogOffset {
                 next_offset: self.next_offset.load(Ordering::Relaxed),
             })
             .await
             .context("Failed to fetch log entries")?
             .into_inner();
-
-        let frames = frames
-            .map(|r| {
-                r.context("frame stream")
-                    .and_then(|f| Frame::try_from_bytes(f.data.into()))
+        let frames = frames.frames.iter()
+            .map(|f| {
+                Frame::try_from_bytes(f.data.clone())
```

Comment:
An unfortunate clone()... would it help if you called `frames.frames.into_iter()` instead of `frames.frames.iter()`, to own the iterated values?

Reply:
nvm, I'll post a follow-up!